analyzers/src/SonarAnalyzer.VisualBasic/Rules/ArrayPassedAsParams.cs

```diff
@@ -0,0 +1,49 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.VisualBasic;
+
+[DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+public sealed class ArrayPassedAsParams : ArrayPassedAsParamsBase<SyntaxKind, ArgumentSyntax>
+{
+    protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
+
+    protected override SyntaxKind[] ExpressionKinds { get; } =
+        {
+            SyntaxKind.ObjectCreationExpression,
+            SyntaxKind.InvocationExpression
+        };
+
+    protected override ArgumentSyntax GetLastArgumentIfArrayCreation(SyntaxNode expression) =>
+        expression switch
+        {
+            ObjectCreationExpressionSyntax { } creation => CheckLastArgument(creation.ArgumentList),
+            InvocationExpressionSyntax { } invocation => CheckLastArgument(invocation.ArgumentList),
+            _ => null
+        };
+
+    private static ArgumentSyntax CheckLastArgument(ArgumentListSyntax argumentList) =>
+        argumentList is not null
+        && argumentList.Arguments.Any()
+        && argumentList.Arguments.Last().GetExpression() is ArrayCreationExpressionSyntax invocationArray
+        && invocationArray.Initializer is CollectionInitializerSyntax { Initializers.Count: > 0 }
+        ? argumentList.Arguments.Last()
+        : null;
```

Comment:
Can be simplified.
```suggestion
        && arguments.Last() is { Expression: ArrayCreationExpressionSyntax { Initializer: CollectionInitializerSyntax { Initializers.Count: > 0 } } } lastArgument
            ? lastArgument
            : null;
```

Reply:
I don't think the VB can be simplified this way, we don't have Expression property as in CS but a GetExpression() method. But this works totally fine for CS 👍

Reply:
Your VB version is fine. There are some strange [argument kinds in VB](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.visualbasic.syntax.argumentsyntax?view=roslyn-dotnet-4.3.0).